ci: smoke tests failing due to poseidon changes

**Describe the bug**
This morning the Smoke Test workflow started failing on main. I believe this correlates with changes to the poseidon deps in https://github.com/penumbra-zone/poseidon377/pull/34/. 

**To Reproduce**
Steps to reproduce the behavior:
1. Run `docker-compose build` locally, observe failure
2. Or view online workflow here: https://github.com/penumbra-zone/penumbra/actions/runs/3950788991/jobs/6763795372



**Expected behavior**
Dependencies compile just fine.


**Additional context**
I can reproduce the error locally, by running `docker-compose build`, but the raw cli command passes for me locally, outside the container context: `cargo build --release --bin pd --target $(uname -m)-unknown-linux-gnu`. 


<details>
<summary>View error message from build</summary>


```
Compiling tonic-web v0.4.0
error[E0432]: unresolved imports `ark_sponge::constraints`, `ark_sponge::poseidon::constraints`
 --> /usr/local/cargo/git/checkouts/poseidon377-136696cbc1ace871/e6f1cb8/poseidon377/src/r1cs.rs:3:5
  |
3 |     constraints::CryptographicSpongeVar, poseidon::constraints::PoseidonSpongeVar,
  |     ^^^^^^^^^^^                                    ^^^^^^^^^^^ could not find `constraints` in `poseidon`
  |     |
  |     could not find `constraints` in `ark_sponge`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `poseidon377` due to previous error
warning: build failed, waiting for other jobs to finish...
Error: building at STEP "RUN cargo build --release --bin pd --target $(uname -m)-unknown-linux-gnu": while running runtime: exit status 101
ERROR: Service 'pd-node0' failed to build : Build failed
```
</details>

--- a/crypto/Cargo.toml
+++ b/crypto/Cargo.toml
@@ -15,8 +15,8 @@ penumbra-tct = { path = "../tct/", features = ["r1cs"] }
 # Git deps
 decaf377 = { git = "https://github.com/penumbra-zone/decaf377", features = ["r1cs"] }
 decaf377-rdsa = { version = "0.5", git = "https://github.com/penumbra-zone/decaf377-rdsa" }
-poseidon377 = { git = "https://github.com/penumbra-zone/poseidon377", branch = "main", features = ["r1cs"] }
-poseidon-paramgen = { git = "https://github.com/penumbra-zone/poseidon377", branch = "main" }
+poseidon377 = { git = "https://github.com/penumbra-zone/poseidon377", rev = "a2d8c7a3288e2e877ac88a4d8fd3cc4ff2b52c04", features = ["r1cs"] }
+poseidon-paramgen = { git = "https://github.com/penumbra-zone/poseidon377", rev = "a2d8c7a3288e2e877ac88a4d8fd3cc4ff2b52c04" }
 jmt = { git = "https://github.com/penumbra-zone/jmt", branch = "upstream-ics23" }
 f4jumble = { git = "https://github.com/zcash/librustzcash", rev = "2425a0869098e3b0588ccd73c42716bcf418612c" }
 

--- a/tct/Cargo.toml
+++ b/tct/Cargo.toml
@@ -18,7 +18,7 @@ serde = { version = "1.0", features = ["derive", "rc"] }
 parking_lot = "0.12"
 ark-ff = "0.3"
 ark-serialize = "0.3"
-poseidon377 = { git = "https://github.com/penumbra-zone/poseidon377", branch = "main" }
+poseidon377 = { git = "https://github.com/penumbra-zone/poseidon377", rev = "a2d8c7a3288e2e877ac88a4d8fd3cc4ff2b52c04" }
 decaf377 = { git = "https://github.com/penumbra-zone/decaf377", features = ["r1cs"] }
 tracing = { version = "0.1" }
 async-trait = { version = "0.1" }

--- a/transaction/Cargo.toml
+++ b/transaction/Cargo.toml
@@ -14,7 +14,7 @@ penumbra-tct = { path = "../tct" }
 # Git deps
 decaf377 = { git = "https://github.com/penumbra-zone/decaf377" }
 decaf377-rdsa = { git = "https://github.com/penumbra-zone/decaf377-rdsa" }
-poseidon377 = { git = "https://github.com/penumbra-zone/poseidon377", branch = "main" }
+poseidon377 = { git = "https://github.com/penumbra-zone/poseidon377", rev = "a2d8c7a3288e2e877ac88a4d8fd3cc4ff2b52c04" }
 ibc = "0.23.0"
 ibc-proto = { version = "0.22.0", features = ["std"] }
 

